#4068 - Ability to change interface language (i18n)

awachatilf opened the pull request:
**What's in the PR**
* Draft of translation support

**How to test manually**
* Configure the application settings.properties file on french language (locale=fr) and verify that texts are in french

**Automatic testing**
* [ ] PR includes unit tests

**Documentation**
* [ ] PR updates documentation

**Todo**
* [x] Move the strings extracted from the telemetry configuration page back into the HTML files and translate the entire HTML files - to avoid having to disable HTML escaping on the resource models

bHussonAtilf committed:
add WicketApplicationBase_fr.properties

awachatilf committed:
Traduction des fichiers properties du package inception-ui-tagset

awachatilf committed:
Traduction des fichiers properties du package inception-ui-search

awachatilf committed:
Traduction des fichiers properties du package inception-api-annotation

awachatilf committed:
Traduction des fichiers properties du package inception-curation

awachatilf committed:
Traduction des fichiers properties du package inception-curation

awachatilf committed:
Traduction des fichiers properties du package inception-telemetry

awachatilf committed:
Traduction plus appropriée

awachatilf committed:
Traduction des fichiers properties du package inception-ui-annotation

awachatilf committed:
Meilleure traduction

awachatilf committed:
Traduction des fichiers properties du package inception-ui-core

awachatilf committed:
Traduction des fichiers properties du package inception-ui-curation

awachatilf committed:
Traduction des fichiers properties du package inception-ui-dashboard

awachatilf committed:
Traduction des fichiers properties du package inception-ui-external-search

awachatilf committed:
Meilleure traduction

awachatilf committed:
Traduction des fichiers properties du package inception-ui-kb

awachatilf committed:
Traduction des fichiers properties du package inception-ui-project

awachatilf committed:
Traduction des fichiers properties du package inception-ui-project

awachatilf committed:
Traduction des éléments de menu

awachatilf committed:
Traduction des retours de fonction "getLabel"

awachatilf committed:
Entrées dupliquées supprimées

awachatilf committed:
Traductions des fichiers properties du module inception-active-learning

awachatilf committed:
Traductions des fichiers properties du module inception-brat-editor

awachatilf committed:
Traductions des fichiers properties du module inception-concept-linking

awachatilf committed:
Traductions des fichiers properties du module inception-concept-linking

awachatilf committed:
Traductions des fichiers properties du module inception-diam-editor

awachatilf committed:
Traductions des fichiers properties du module inception-diam-editor

awachatilf committed:
Traductions des fichiers properties du module inception-external-search-elastic

awachatilf committed:
Traductions des fichiers properties du module inception-external-search-elastic

awachatilf committed:
Traductions des fichiers properties du module inception-external-search-opensearch

awachatilf committed:
Traductions des fichiers properties du module inception-external-search-opensearch

awachatilf committed:
Traductions des fichiers properties du module inception-external-search-pubannotation

awachatilf committed:
Traductions des fichiers properties du module inception-external-search-pubannotation

awachatilf committed:
Traductions des fichiers properties du module inception-external-search-solr

awachatilf committed:
Traductions des fichiers properties du module inception-external-search-solr

awachatilf committed:
Traductions des fichiers properties du module inception-feature-lookup

awachatilf committed:
Traductions des fichiers properties du module inception-feature-lookup

awachatilf committed:
Traductions des fichiers properties du module inception-guidelines

awachatilf committed:
Traductions des fichiers properties du module inception-guidelines

awachatilf committed:
Traductions des fichiers properties du module inception-imls-elg

awachatilf committed:
Traductions des fichiers properties du module inception-imls-elg

awachatilf committed:
Traductions des fichiers properties du module inception-imls-external

awachatilf committed:
Traductions des fichiers properties du module inception-imls-external

awachatilf committed:
Traductions des fichiers properties du module inception-agreement

awachatilf committed:
Traductions des fichiers properties du module inception-agreement

awachatilf committed:
Traductions des fichiers properties du module inception-workload-ui

awachatilf committed:
Traductions des fichiers properties du module inception-workload-ui

awachatilf committed:
Traductions des fichiers properties du module inception-workload-matrix

awachatilf committed:
Traductions des fichiers properties du module inception-workload-matrix

awachatilf committed:
Traductions des fichiers properties du module inception-workload-dynamic

awachatilf committed:
Traductions des fichiers properties du module inception-workload-dynamic

awachatilf committed:
Traductions des fichiers properties du module inception-versioning

awachatilf committed:
Traductions des fichiers properties du module inception-versioning

awachatilf committed:
Traductions des fichiers properties du module inception-ui-agrement

awachatilf committed:
Traductions des fichiers properties du module inception-ui-agrement

awachatilf committed:
Traductions des fichiers properties du module inception-imls-hf

awachatilf committed:
Traductions des fichiers properties du module inception-imls-hf

awachatilf committed:
Traductions des fichiers properties du module inception-imls-lapps

awachatilf committed:
Traductions des fichiers properties du module inception-imls-lapps

awachatilf committed:
Traductions des fichiers properties du module inception-imls-weblicht

awachatilf committed:
Traductions des fichiers properties du module inception-imls-weblicht

awachatilf committed:
Traductions des fichiers properties du module inception-imls-opennlp

awachatilf committed:
Traductions des fichiers properties du module inception-imls-opennlp

awachatilf committed:
Traductions des fichiers properties du module inception-kb-fact-linking

awachatilf committed:
Traductions des fichiers properties du module inception-kb-fact-linking

awachatilf committed:
Traductions des fichiers properties du module inception-layer-docmetadata

awachatilf committed:
Traductions des fichiers properties du module inception-layer-docmetadata

awachatilf committed:
Traductions des fichiers properties du module inception-project-export

awachatilf committed:
Traductions des fichiers properties du module inception-project-export

awachatilf committed:
Traductions des fichiers properties du module inception-recommendation

awachatilf committed:
Traductions des fichiers properties du module inception-recommendation

awachatilf committed:
Traductions des fichiers properties du module inception-recommendation-api

awachatilf committed:
Traductions des fichiers properties du module inception-recommendation-api

awachatilf committed:
Traductions des fichiers properties du module inception-security

awachatilf committed:
Traductions des fichiers properties du module inception-security

awachatilf committed:
Traductions des fichiers properties du module inception-security

awachatilf committed:
Traductions des fichiers properties du module inception-security

awachatilf committed:
Traductions des fichiers properties du module inception-sharing

awachatilf committed:
Traductions des fichiers properties du module inception-sharing

awachatilf committed:
Traductions des fichiers properties du module inception-support

awachatilf committed:
Traductions des fichiers properties du module inception-support

awachatilf committed:
Traductions des fichiers properties du module inception-support-standalone

awachatilf committed:
Traductions des fichiers properties du module inception-support-standalone

awachatilf committed:
Fichiers properties par module (partiellement réalisé)

awachatilf committed:
Fichiers properties par module (partiellement réalisé)

awachatilf committed:
L'implémentation des fichiers properties par module ne semble pas fonctionner sur toutes les pages (retour à l'implémentation précédente)

awachatilf committed:
L'implémentation des fichiers properties par module ne semble pas fonctionner sur toutes les pages (retour à l'implémentation précédente)

awachatilf committed:
Ordre des entrées dans les fichiers properties

awachatilf committed:
Ordre des entrées dans les fichiers properties

awachatilf committed:
On fait appel à un fichier de traductions du module pour la classe MatrixWorkloadManagementPageMenuItem

awachatilf committed:
On fait appel à un fichier de traductions du module pour la classe MatrixWorkloadManagementPageMenuItem

awachatilf committed:
La traduction modularisée ne semble pas fonctionner sur la page d'administration donc on retourne en arrière

awachatilf committed:
La traduction modularisée ne semble pas fonctionner sur la page d'administration donc on retourne en arrière

reckart committed:
#4058 - Cannot export texts containing certain characters as UIMA CAS XMI

- Rename packages in XMI support module
- Switch to modern spring factories declaration

reckart committed:
- Rename packages in XMI support module
- Switch to modern spring factories declaration

awachatilf committed:
Implémentation des fichiers properties par module pour les items de menu

awachatilf committed:
Implémentation des fichiers properties par module pour les items de menu

awachatilf committed:
Implémentation des fichiers properties par module pour les ProjectPageMenuItem

awachatilf committed:
Implémentation des fichiers properties par module pour les ProjectPageMenuItem

awachatilf committed:
Implémentation des fichiers properties par module pour les items de menu

awachatilf committed:
Implémentation des fichiers properties par module pour les items de menu

awachatilf committed:
Implémentation des fichiers properties par module pour les items de menu

awachatilf committed:
Implémentation des fichiers properties par module pour les items de menu

awachatilf committed:
Traductions de fichiers HTML de télémétrie

awachatilf committed:
Traductions de fichiers HTML de télémétrie

ukp-svc-jenkins commented:
Can one of the admins verify this patch?

reckart committed:
Merge branch 'main' into feature/4068-i18n-support

reckart commented on inception/inception-curation/src/main/java/de/tudarmstadt/ukp/inception/curation/settings/CurationProjectSettingsMenuItem.java:
```diff
@@ -50,7 +53,13 @@ public IconType getIcon()
     public String getLabel()
     {
-        return "Curation";
+        IRequestablePage currentPage = PageRequestHandlerTracker.getLastHandler(RequestCycle.get())
```
We have a new static method `de.tudarmstadt.ukp.clarin.webanno.support.wicket.resource.Strings.getString(String)` that should be used in backend code like this. However, it requires that the string to be obtained is contained in a `wicket-package.properties` file in the package or parent-package of the code that calls the method.

I am using this new method in very few places at the moment, e.g. in `de.tudarmstadt.ukp.inception.annotatorjs.AnnotatorJsHtmlAnnotationEditorFactory.getDisplayName()`.

reckart commented on inception/inception-telemetry/src/main/java/de/tudarmstadt/ukp/clarin/webanno/telemetry/matomo/MatomoTelemetryTraitsEditor_fr.html:
```diff
@@ -0,0 +1,66 @@
+<!--
```
I wonder if it would be better to pull out the strings here and put them into a `.properties` file instead of having multiple versions of the `.html` files around.

reckart replied:
Thanks for trying this out. Turns out it was a bad idea because the extracted strings contain HTML markup which would force use disabling HTML escaping on the resource models - which we do not want to do. So better to translate the entire HTML files as you had. I have added a corresponding todo to the initial description above.

reckart committed:
#3673 - Update dependencies

- DKPro Core 2.3.1 -> 2.4.0
- RDf4J 4.2.3 -> 4.3.2

awachatilf committed:
Suppression des imports en .*

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - Removal of .* imports

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - Removal of .* imports

awachatilf committed:
Utilisation de la fonction Strings.getString() pour ManageUsersPageMenuItem

awachatilf committed:
Utilisation de la fonction Strings.getString() pour ManageUsersPageMenuItem

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - Strings.getString() function usage for ManageUsersPageMenuItem

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - Strings.getString() function usage for ManageUsersPageMenuItem

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - Strings.getString() function use for ManageUsersPageMenuItem

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - Strings.getString() function use for ManageUsersPageMenuItem

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - Strings.getString() function use for ManageUsersPageMenuItem

awachatilf committed:
- Strings.getString() function usage for ManageUsersPageMenuItem

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - Strings.getString() function usage for ManageUsersPageMenuItem

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - Strings.getString() function usage for ManageUsersPageMenuItem

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - Strings.getString() function usage for ManageUsersPageMenuItem

reckart committed:
#4058 - Cannot export texts containing certain characters as UIMA CAS XMI

- Enable sanitziing of illegal characters on export for XMI formats
- Added option to control if sanitation happens or not
- Updated documentation of XMI formats

reckart committed:
#4058 - Cannot export texts containing certain characters as UIMA CAS XMI

- Enable sanitziing of illegal characters on export for XMI formats
- Added option to control if sanitation happens or not
- Updated documentation of XMI formats

reckart committed:
No issue: Experiment with a Jenkinsfile

reckart committed:
No issue: Experiment with a Jenkinsfile

reckart committed:
No issue: Experiment with a Jenkinsfile

reckart committed:
No issue: Experiment with a Jenkinsfile

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - Use Strings.getString() function for translations

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - Use Strings.getString() function for translations

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - Use Strings.getString() function for translations

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - ManageUsersPageMenuItem label value in WicketApplicationBase.properties file

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - ManageUsersPageMenuItem label value in WicketApplicationBase.properties file

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - ManageUsersPageMenuItem label value in WicketApplicationBase.properties file

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - ManageUsersPageMenuItem label value in WicketApplicationBase.properties file

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - wicket-package.properties translation files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - wicket-package.properties translation files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - wicket-package.properties translation files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - better translation ("layer" to "couche")

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - better translation ("layer" to "couche")

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - better translation ("layer" to "couche")

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - better translation

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - better translation

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - better translation

reckart committed:
No issue: Experiment with a Jenkinsfile

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - Telemetry HTML pages translation with properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - Telemetry HTML pages translation with properties files

awachatilf committed:
- Telemetry HTML pages translation with properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - Telemetry HTML pages translation with properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - Telemetry HTML pages translation with properties files

awachatilf committed:
- Telemetry HTML pages translation with properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - Telemetry HTML pages translation with properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - Telemetry HTML pages translation with properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - Telemetry HTML pages translation with properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - Telemetry HTML pages translation with properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - Telemetry HTML pages translation with properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - Telemetry HTML pages translation with properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - Telemetry HTML pages translation with properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - Telemetry HTML pages translation with properties files

awachatilf commented on inception/inception-telemetry/src/main/java/de/tudarmstadt/ukp/clarin/webanno/telemetry/TelemetryStrings.java:
I've tried transferring the translation of the HTML files to the properties files in order to centralise the translations in wicket.properties file as it was suggested before. I've encountered the problem of variable interpolation within the text (such as "product.name" or "save"). I wanted to know if my approach was the right one by creating this class (TelemetryStrings) and naming it this way in order to call the variables in the properties file using the notation ${variable}. Could someone tell me if I did it in the proper way?

reckart replied:
Thanks for the suggestion. It looks a bit strange. The question is why the properties are not interpolated - IMHO that should just work.

reckart replied:
Another consideration - if we need to set `setEscapeModelStrings(false)` in order to pull the strings out of the HTML files, then I'd say it may be better to internationalize the entire files as you had before.

reckart replied:
Ok, so the correct syntax for properties inside properties is `#(propertyname)`. I have removed the TelemetryStrings and fixed the properties files - now it works as expected.

reckart committed:
No issue. Slightly improving PDF editor visual

reckart committed:
Merge branch 'main' into feature/4068-i18n-support

reckart committed:
#4068 - Ability to change interface language

- Remove TelemetryStrings class again
- Use proper nested properties syntax

reckart committed:
#4068 - Ability to change interface language

- Remove TelemetryStrings class again
- Use proper nested properties syntax

reckart committed:
#4068 - Ability to change interface language

- Remove TelemetryStrings class again
- Use proper nested properties syntax

reckart committed:
#4068 - Ability to change interface language

- Remove TelemetryStrings class again
- Use proper nested properties syntax

reckart commented:
@awachatilf Looks like this PR is coming along pretty well. I wonder what (if anything) would be missing from your side before we could merge it?

The commit messages in the PR do not match the project convention. We have two options here:

1) you rewrite the commit comments using an interactive git rebase. Mind that depending on what tool you are using to do this, you may have to change the comment character option in git temporarily to something other than `#` because our initial comment line always starts with `#NUMBER - TITLE`...
2) we squash-merge the PR. All commits will become a single commit in this process and I rewrite the commit comment at merge time

I saw that some of the newly introduced properties match what seems should be a "core term", e.g. 👍 

```
curationprojectsettings.menuitem.label=Curation
curationprojectsettings.panel.factory.label=Curation
```

Instead of introducing new properties here, I would have tried introducing a "core term" `curation=Curation` and put that into `WicketApplicationBase.properties` - or at least not introduce multiple properties for essentially the same term. But it is a rather philosophical comment. Having strictly separate properties for every single string is also an approach and has its own benefits. I do not have a particularly strong opinion about either approach at this point.

The English properties files are in UTF-8. The French ones are in ISO-8859-1. I believe it would be good to use UTF-8 for both of them. For this to work properly, we should rename them from `.properties` to `.utf8.properties` though. But let's maybe do that in a separate PR. This one is already getting very very large - and too large PRs are not good.

reckart committed:
#4068 - Ability to change interface language

- Remove commented out lines
- Restore empty lines

reckart committed:
#4068 - Ability to change interface language

- Remove commented out lines
- Restore empty lines

reckart committed:
#4068 - Ability to change interface language

- Remove commented out lines
- Restore empty lines

awachatilf committed:
Merge remote-tracking branch 'origin/feature/4068-i18n-support' into feature/4068-i18n-support

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - Telemetry HTML pages translation without HTML escaping

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - Telemetry HTML pages translation without HTML escaping

awachatilf commented on inception/inception-telemetry/src/main/java/de/tudarmstadt/ukp/clarin/webanno/telemetry/ui/TelemetrySettingsPage.html:
```diff
@@ -25,13 +25,10 @@
-                there were any changes to the data collection process, this page will appear again
-                when an administrator logs in. You can always revise your choices by accessing this
-                page via the <i class="fa fa-heartbeat"></i> icon in the page footer.
+                <wicket:message key="settingsIntro">
```
I think I've found a way to use a translation properties file with HTML content without escaping HTML. I created wicket elements for texts inside tags (like <b>) and translated them separately. Do you think this is the right way to do it? I'd prefer all the translations to remain centralised in the properties files because this will be less complicated to maintain translations later on.

reckart replied:
Actually, that is not necessary. When embedding strings in a page using `wicket:message`, then the strings in the properties files can contain markup. Only if the strings are included through a `ResourceModel`, we would need to turn off escaping to support markup. In fact, we should try to work with `wicket:message` as much as possible and use `ResourceModel` only for cases where the text in a component needs to change dynamically (i.e. where the message key is not known or not fixed).

reckart replied:
I made a commit to change the Telemetry page accordingly.

reckart committed:
#4068 - Ability to change interface language (i18n)

- Simplify telemetry strings

reckart committed:
#4068 - Ability to change interface language (i18n)

- Simplify telemetry strings

reckart commented:
> There are a lot of hardcoded strings in java files and I'm not sure which one should be translated or not. For example, I suppose logger message shouldn't, but I suppose corresponding functions of Wicket Component class (success(), info(), warn(), error() and fatal()) that seem to display popup messages should. For some other functions like getName(), getDescription() or getDisplayname() of some interfaces I'm not really sure if they should be translated. I'm not sure either if Validate.notNull() strings are displayed in the application and should be translated. There are many more, but these seems to be most common cases.

We should for the moment focus on the properties files that are already there and not go through all the Java files (and possibly typescript files) to fix everything. Translating only the properties files we already have will already be a big step. After that, we might go through the code module by module and see if/which strings in Java files should be internationalized. But that should be a separate issue and then best separate PRs for different modules.

In any case, we won't be able to translate *everything*  because many strings are not even under our control (they come from third-party libraries or from the JVM itself).

reckart commented:
> I should rename properties files key with shorter names as you told me to. (I don't know if you saw Benjamin Husson's proposition of prefixing package keys with local and application ones with global (or something like))

Also not critical. This can happen later. I still would like to make use of some tooling that generates Java files with constants for the keys so we can a compiler-checkable connection to the strings. Latest at that point, we'll notice which strings have awkward (longish) names and can fix them.

reckart commented:
> Are you sure english properties file are currently in UTF-8? The "ä" in "Universität" first line comments doesn't seem to display well in UTF-8. In any case, if it's possible to encode in UTF-8 I also think it's better. But if you prefer me to do that in another PR there's no problem.

Yes, let's do that separately. Separate issue as well.

awachatilf committed:
- inception-ui-core HTML files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-ui-core HTML files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-ui-dashboard HTML files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-ui-dashboard HTML files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-ui-annotation HTML files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-ui-annotation HTML files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-ui-curation HTML files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-ui-curation HTML files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-ui-project HTML files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-ui-project HTML files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-ui-search HTML files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-ui-search HTML files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-ui-external-search HTML files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-ui-external-search HTML files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-ui-tagsets files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-ui-tagsets files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-api-annotation files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-api-annotation files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-imls-elg files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-imls-elg files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-imls-hf files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-imls-hf files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-imls-weblicht files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-imls-weblicht files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-curation files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-curation files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-project-export files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-project-export files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-pdf-editor files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-pdf-editor files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-pdf-editor2 files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-pdf-editor2 files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-recommendation files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-recommendation files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-active-learning files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-active-learning files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-diam-editor files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-diam-editor files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-image files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-image files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-kb-fact-linking files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-kb-fact-linking files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-layer-docmetadata files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-layer-docmetadata files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-sharing files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-sharing files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-support files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-support files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-support files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-support files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-workload-dynamic files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-workload-dynamic files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-workload-matrix files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - inception-workload-matrix files translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - placeholders texts translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - placeholders texts translation using properties files

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - better translation

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - better translation

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - better translations

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - better translations

awachatilf committed:
- translations of title HTML attributes

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - translations of title HTML attributes

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - better translation

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - better translation

awachatilf committed:
Merge remote-tracking branch 'origin/feature/4068-i18n-support' into feature/4068-i18n-support

# Conflicts:
#	inception/inception-api-annotation/src/main/java/de/tudarmstadt/ukp/clarin/webanno/api/annotation/preferences/PreferencesActionBarItem.html
#	inception/inception-guidelines/src/main/java/de/tudarmstadt/ukp/inception/guidelines/GuidelinesActionBarItem.html
#	inception/inception-ui-core/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/core/WicketApplicationBase.properties
#	inception/inception-ui-core/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/core/WicketApplicationBase_fr.properties
#	inception/inception-ui-core/src/main/java/de/tudarmstadt/ukp/inception/ui/core/menubar/MenuBar_fr.properties
#	inception/inception-workload-matrix/src/main/java/de/tudarmstadt/ukp/inception/workload/matrix/annotation/MatrixWorkflowActionBarItemGroup.html

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - missing accent in translation

awachatilf committed:
Merge branch 'refs/heads/main' into feature/4068-i18n-support

# Conflicts:
#	inception/inception-processing/src/main/java/de/tudarmstadt/ukp/inception/processing/BulkProcessingPageMenuItem.java
#	inception/inception-security/src/main/java/de/tudarmstadt/ukp/inception/security/client/auth/apikey/ApiKeyAuthenticationTraitsEditor.properties
#	inception/inception-support-bootstrap/src/main/java/de/tudarmstadt/ukp/inception/bootstrap/dialog/ChallengeResponseDialog_fr.properties
#	inception/inception-support-bootstrap/src/main/java/de/tudarmstadt/ukp/inception/bootstrap/dialog/ConfirmationDialog_fr.properties
#	inception/inception-support/src/main/java/de/tudarmstadt/ukp/inception/support/wicket-package.properties
#	inception/inception-support/src/main/java/de/tudarmstadt/ukp/inception/support/wicket-package_fr.properties
#	inception/inception-ui-agreement/src/main/java/de/tudarmstadt/ukp/inception/ui/agreement/page/AgreementPage.properties
#	inception/inception-ui-annotation/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/annotation/AnnotationPageMenuItem.java
#	inception/inception-ui-core/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/core/WicketApplicationBase.properties
#	inception/inception-ui-core/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/core/footer/VersionFooterPanel.html
#	inception/inception-ui-core/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/core/login/LoginPage.html
#	inception/inception-ui-curation/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/curation/page/CurationPage.html
#	inception/inception-ui-dashboard-activity/src/main/java/de/tudarmstadt/ukp/inception/ui/core/dashboard/activity/panel/ActivityPanel.html
#	inception/inception-ui-dashboard/src/main/java/de/tudarmstadt/ukp/inception/ui/core/dashboard/projectlist/ProjectsOverviewPage.html
#	inception/inception-ui-kb/src/main/java/de/tudarmstadt/ukp/inception/ui/kb/wicket-package.properties
#	inception/inception-ui-project/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/project/detail/ProjectDetailPanel.html
#	inception/inception-ui-project/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/project/layers/ProjectLayersPanel.html
#	inception/inception-ui-tagsets/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/tagsets/TagEditorPanel.html
#	inception/inception-workload-dynamic/src/main/java/de/tudarmstadt/ukp/inception/workload/dynamic/DynamicWorkloadExtensionImpl.java
#	inception/inception-workload-matrix/src/main/java/de/tudarmstadt/ukp/inception/workload/matrix/MatrixWorkloadExtensionImpl.java
#	inception/inception-workload-matrix/src/main/java/de/tudarmstadt/ukp/inception/workload/matrix/annotation/MatrixWorkflowActionBarItemGroup.html
#	inception/inception-workload-matrix/src/main/java/de/tudarmstadt/ukp/inception/workload/matrix/management/MatrixWorkloadFilterPanel.html
#	inception/inception-workload-matrix/src/main/java/de/tudarmstadt/ukp/inception/workload/matrix/management/MatrixWorkloadManagementPage.html
#	inception/inception-workload-matrix/src/main/java/de/tudarmstadt/ukp/inception/workload/matrix/management/MatrixWorkloadManagementPage.properties

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - Update from upstream branch

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - Login page translation

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - System status dashlet page translation

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - Log messages page translation

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - Project overview page translation

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - Project layers panel translation

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - Predefined layers translation

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - Predefined new projects translation

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - Activities dashlet translation

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - Toggle state translation

awachatilf committed:
#4068 - Ability to change interface language (i18n)

 - Various translation

reckart commented:
Closing due to inactivity. I'll try looking into an alternative solution.
